src/lib/sql/sql_translator.cpp

```diff
@@ -355,11 +356,21 @@ std::shared_ptr<AbstractLQPNode> SQLTranslator::_translate_join(const hsql::Join
   auto left_node = _translate_table_ref(*join.left);
   auto right_node = _translate_table_ref(*join.right);
 
-  const hsql::Expr& condition = *join.condition;
+  std::vector<const hsql::Expr*> condition_list;
+  _flatten_conjunctive_expression_tree(join.condition, condition_list);
+
+  auto get_side =
+      std::bind(&SQLTranslator::_get_side, this, std::cref(left_node), std::cref(right_node), std::placeholders::_1);
+  std::vector<const hsql::Expr*> joins, left, right;
+  _get_sides_from_operator_expressions(condition_list, get_side, joins, left, right);
```

Comment:
this selects the subplan a condition should be applied to. Call it `_assign_join_predicates_to_subplans` or something. the current name is just too vague ("side" is the wrong word since you're also assigning to the join condition, "operator expressions" could be anything, but you mean "join predicates")

Reply:
You'll need to realise that `_get_sides_from_operator_expressions` is no longer concerned with join predicates at all.
The fact that this particular method is used with join conditions is not relevant to its execution nor its guarantees. 

> "side" is the wrong word since you're also assigning to the join condition,

By coincidence a "join condition" is a condition involving column references of both join partners. To rephrase, above code is stating: in order to get hold of join conditions, call a routine that collects `hsql::Expr` with column references of both sides.

> "operator expressions" could be anything, but you mean "join predicates"

Correct me if I'm wrong, I think any `hsql::Expr` of type `kExprOperator` can combine two  expressions of type `kExprColumnRef`.

I think it's possible to reconstruct why the naming was chosen like this. I currently see no good reason why the subjects and names of `_translate_join` should leak into called methods of more general kind.

Reply:
I think I get where you're coming from. `_get_sides_from_operator_expressions` is supposed to be a "general" method that can work on any list of `hsql::Exprs` and any `get_side` functor?

I have no problem with "general purpose" methods that can fulfill more than just one task, the question is just: where else would you use this method, what justifies making it a more abstract and therefore harder to understand?

Reply:
Apparently naming is a hard thing ;)
With the method as of current state (functor is gone anyway, which probably attracted of most of the criticism) I believe it to be understandable. Choosing a name including terms like "join" etc. would give hints about how it is used (caller perspective) and give poor hints about what is accomplished particular to this method (without understanding its (only) usage first).

I do like discussions about code-reuse. Probably the question on "where else would you *actually* use this X?" remains unanswered 99% of the time at the design-time of the thing-to-be-reused. The bottom line is, if one does not design for reuse, it is never going to happen anyway.